cardano-cli/src/Cardano/CLI/Compatible/Exception.hs

```diff
@@ -41,31 +42,37 @@ instance Exception CustomCliException where
 -- | Wrapper function which allows throwing of types of instance `Error`, attaching call stack
 -- from the call site
 throwCliError
-  :: forall e m a
-   . (HasCallStack, Show e, Typeable e, Error e, MonadIO m)
+  :: (HasCallStack, Show e, Typeable e, Error e, MonadIO m)
   => e
   -> m a
 throwCliError = withFrozenCallStack $ throwIO . CustomCliException
 
 fromEitherCli
-  :: forall e m a
-   . (HasCallStack, MonadIO m, Show e, Typeable e, Error e)
+  :: (HasCallStack, MonadIO m, Show e, Typeable e, Error e)
```

Comment:
Those `forall`s were here on purpose. Now the order of type variables is sub-optimal. The problem here is that the previous version was allowing to use `TypeApplications` to enforce the error type i.e.
```haskell
fromEitherCli @(FileError ()) foo
```
Now this version can't do that.
```suggestion
  :: (HasCallStack, Show e, Typeable e, Error e, MonadIO m)
```